ganga/GangaCore/Utility/Virtualization.py

```diff
@@ -0,0 +1,64 @@
+import subprocess
+import os
+
+
+def checkSingularity():
+    """Check whether Singularity is installed and the current user has right to access
+
+        Return value: True or False"""
+
+    nullOutput = open(os.devnull, 'wb')
+    returnCode = 1
+    try:
+        returnCode = subprocess.call(["singularity", "--version"], stdout=nullOutput, stderr=nullOutput)
+    except:
+        pass 
+    if returnCode == 0 : return True
+    return False
+
+
+def checkDocker():
+    """Check whether Docker is installed and the current user has right to access
+
+        Return value: True or False"""
+
+    nullOutput = open(os.devnull, 'wb')
+    returnCode = 1
+    try:
+        returnCode = subprocess.call(["docker", "ps"], stdout=nullOutput, stderr=nullOutput)
+    except:
+        pass 
+    if returnCode == 0 : return True
+    return False
+
+
+def checkUDocker():
+    """Check whether UDocker is installed and the current user has right to access
+
+        Return value: True or False"""
+
+    fname = os.path.join(os.path.expanduser("~"),"udocker")
+    nullOutput = open(os.devnull, 'wb')
+    if (os.path.isfile(fname)):
+        returnCode = subprocess.call([fname, "ps"], stdout=nullOutput, stderr=nullOutput)
+        if (returnCode == 0):
+            return True
+    return False
+
+
+def installUdocker():
+    """Download and install UDocker
+
+        Return value: True (If Success) or False"""
+
+    fname = os.path.join(os.path.expanduser("~"),"udocker")
```

Comment:
As @egede points out, this may not work on workernode. I assume we change it to
```suggestion
    fname = os.path.join(os.getcwd(), "udocker")
```

Reply:
I think we might just need to let the installation location be dependent on the backend. For the Local backend it makes sense to install to a shared location to facilitate re-use. It should probably be a configurable though. An even bigger deal in fact is the location of the images that are downloaded. By default they go into `~/.udocker`

Reply:
ah sorry your right I though I was only looking at Dirac (non local) code.

Reply:
Locations of both the script and the images is now configurable.